### PR TITLE
feat(chat-panel): add vscode bundle script

### PR DESCRIPTION
**Pull request by wwayne**
### Context
In the VSCode chat panel, we generate a Webview using the following code snippet. It consists of HTML that loads a js script.
```js
const agent = await createAgentInstance(this._context);
const { server } = agent.getConfig()
const scriptUri = getUri(webview, extensionUri, ['chat-panel', "index.js"]);
return `
      <!DOCTYPE html>
      <html lang="en">
        <head>
          <meta charset="UTF-8" />
          <meta name="viewport" content="width=device-width, initial-scale=1.0" />
          <title>Tabby</title>
        </head>
        <body>
          <div id="root"></div>
          <script>window.endpoint="${server.endpoint}"</script>
          <script>window.token="${server.token}"</script>
          <script type="module" src="${scriptUri}"></script>
        </body>
      </html>
`;
```

### Goal
In `clients/tabby-chat-panel`, it can generate a bundled JS file used for the VSCode webview.
1. In `clients/tabby-chat-panel`, command `yarn build:vscode` will generate a bundled JS file into clients/vscode/chat-panel/
2. In `clients/vscode`, command `yarn build` will call `clients/tabby-chat-panel` to generate the JS file

### CODE REVIEW

File: clients/tabby-chat-panel/src/react.ts [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
 import { createClient, createServer } from './index'
 
 function useClient(iframeRef: RefObject<HTMLIFrameElement>, api: ClientApi) {
   const clientRef = useRef<ServerApi | null>(null)
 
   useEffect(() => {
-    if (iframeRef.current && !clientRef.current) {
+    if (iframeRef.current && !clientRef.current)
       clientRef.current = createClient(iframeRef.current, api)
-    }
```
Comment (wsxiaoys) [inline clients/tabby-chat-panel/src/react.ts:13]: Please revert the change. always use `{}` with `if`

**Comment by wsxiaoys** [comment]
cc @icycodes

File: clients/tabby-chat-panel/src/react.ts [hunk 2]
```diff
@@ -21,7 +20,8 @@ function useServer(api: ServerApi) {
 
   useEffect(() => {
     const isInIframe = window.self !== window.top
-    if (isInIframe && !serverRef.current) serverRef.current = createServer(api)
+    if (isInIframe && !serverRef.current)
+      serverRef.current = createServer(api)
```
Comment (wsxiaoys) [inline clients/tabby-chat-panel/src/react.ts:24]: ditto, use `{}`

File: clients/tabby-chat-panel/src/react.ts
Comment (wsxiaoys) [inline clients/tabby-chat-panel/src/react.ts:1]: ditto, revert changes to this file since they're all formatting related

File: clients/tabby-chat-panel/src/index.ts
Comment (wsxiaoys) [inline clients/tabby-chat-panel/src/index.ts:1]: i suggest revert change to this file as they're formatting changes.

File: clients/tabby-chat-panel/src/vscode.tsx
Comment (wsxiaoys) [inline clients/tabby-chat-panel/src/vscode.tsx:1]: rename this to `iframe.tsx`, since it's likely reuse by jetbrains as well

**Review by wsxiaoys — APPROVED** [review]
Otherwise LGTM

File: clients/tabby-chat-panel/package.json [hunk 3]
```diff
@@ -42,17 +42,21 @@
     "start": "tsx src/index.ts",
     "test": "vitest",
     "test:once": "vitest run",
-    "typecheck": "tsc --noEmit"
+    "typecheck": "tsc --noEmit",
+    "build:vscode": "yarn && esbuild src/vscode.tsx --bundle --minify --outfile=../vscode/chat-panel/index.js --loader:.js=jsx",
```
Comment (wsxiaoys) [inline clients/tabby-chat-panel/package.json:46]: edit build.config.ts

File: clients/tabby-chat-panel/src/react.ts (hunk 1 above)
Comment (wwayne) [reply clients/tabby-chat-panel/src/react.ts]: update the eslint
```
export default antfu({
  rules: {
    curly: ['off', 'all'],
  },
})
```

File: clients/tabby-chat-panel/package.json (hunk 3 above)
Comment (wwayne) [reply clients/tabby-chat-panel/package.json]: left a FIXME comment, because I have no idea how to edit the config file, I don't wanna spend time to check this document at the moment
